packages/shared-ux/modal/tabbed/src/tabbed_modal.tsx

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, {
+  useMemo,
+  useCallback,
+  Fragment,
+  type ComponentProps,
+  type FC,
+  type ReactElement,
+} from 'react';
+import {
+  EuiButton,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiTabs,
+  EuiTab,
+  type EuiTabProps,
+  type CommonProps,
+} from '@elastic/eui';
+import {
+  ModalContextProvider,
+  useModalContext,
+  type ITabDeclaration,
+  type IDispatchFunction,
+  type IModalContextProviderProps,
+} from './context';
+
+export type IModalTabContent<S> = (props: {
+  state: S;
+  dispatch: IDispatchFunction;
+}) => ReactElement;
+
+interface IModalTabActionBtn<S> extends CommonProps {
+  id: string;
+  dataTestSubj: string;
+  label: string;
+  handler: (args: { state: S }) => void;
+  isCopy?: boolean;
+  style?: (args: { state: S }) => boolean;
+}
+
+export interface IModalTabDeclaration<S = {}> extends EuiTabProps, ITabDeclaration<S> {
+  description?: string;
+  'data-test-subj'?: string;
+  content: IModalTabContent<S>;
+  modalActionBtn: IModalTabActionBtn<S>;
```

Comment:
Would be nice to make this `modalActionBtn` optional. I cannot imagine that all tabs will need this.